polkadot/node/core/pvf/src/execute/queue.rs

```diff
@@ -638,3 +653,289 @@ pub fn start(
 	.run();
 	(to_queue_tx, from_queue_rx, run)
 }
+
+struct Unscheduled {
+	unscheduled: HashMap<PvfExecPriority, VecDeque<ExecuteJob>>,
+	counter: HashMap<PvfExecPriority, usize>,
+}
+
+impl Unscheduled {
+	// A threshold reaching which we reset counted jobs.
+	// The max expected queue_size in normal conditions, at the beginning of each block,
+	// a validator will submit jobs for at least vrf_module_samples(6) + 1 for backing
+	// the parachain candidate they are assigned to, there is some buffer added to cover
+	// for situations, where more work arrives in the queue.
+	const MAX_COUNT: usize = 12;
+	// A threshold in percentages, the portion a current priority can "steal" from lower ones.
+	// For example:
+	// Disputes take 70%, leaving 30% for approvals and all backings.
+	// 80% of the remaining goes to approvals, which is 30% * 80% = 24% of the original 100%.
```

Comment:
This way of expressing the percentages is not easy. You have to do this math everytime. Can it be made simpler to not require the reader to do much math ?

Reply:
I tried to update the docs to make it clearer. Unfortunately I didn't come up with a better solution.